### PR TITLE
[Bugs] Improve photos support

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #310, #487 

- Fix regression: photo doesn't load
- Fix bug: remote path contained "N/A" instead of `formId` when creating a new observation instead of editing old one
- Fix bug: bottom sheet dialog not dismissing after making a selection
- Use `UUID` as filename in order to support multiple photos (in upcoming PRs)
- Enqueue photos for upload when synchronizing response deltas
- Remove local file after uploading to remote storage successfully

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
@gino-m PTAL

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  @Override
+  public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+    holder.binding.textView.setText(ADD_PHOTO_LISt[position][0]);
+    holder.binding.imageView.setImageResource(ADD_PHOTO_LISt[position][1]);
+    holder.type = ADD_PHOTO_LISt[position][2];
+  }
+
+  @Override
+  public int getItemCount() {
+    return ADD_PHOTO_LISt.length;
```
Comment (dturner) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java]: Typo. Should be `ADD_PHOTO_LIST` here and throughout.

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import android.view.View.OnClickListener;
+import android.view.ViewGroup;
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+import com.google.android.gnd.R;
+import com.google.android.gnd.databinding.AddPhotoListItemBinding;
+
+public class AddPhotoDialogAdapter extends RecyclerView.Adapter<AddPhotoDialogAdapter.ViewHolder> {
+
+  public static final int CAMERA = 1;
```
Comment (dturner) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java:30]: It's not immediately clear what these constants are for. Maybe change to `PHOTO_SOURCE_CAMERA` and `PHOTO_SOURCE_STORAGE`

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  public static final int STORAGE = 2;
+
+  private static final int[][] ADD_PHOTO_LISt = {
```
Comment (dturner) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java]: Maybe change this to `PHOTO_SOURCE_RESOURCES`. 
 
You could also create a separate inner class e.g. `class PhotoStorageResource { int actionStringResourceId, int iconResourceId, int sourceType }` and instead make this `private static final ArrayList<PhotoStringResource> = { ... }; // list of possible photo sources and their associated resources`

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java (hunk 1 above)
Comment (dturner) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java]: Ah, you've already fixed this (I was reviewing on a per-commit basis so didn't spot it until after I'd made the comment).

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java (hunk 3 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java]: Done

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java (hunk 2 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/AddPhotoDialogAdapter.java]: Done

**Comment by shobhitagarwal1612** [comment]
@dturner PTAL.

**Comment by shobhitagarwal1612** [comment]
@dturner. Gentle ping

**Pull request merged**